### DISCUSSION TITLE
Log4J vulnerability?

### DISCUSSION THREAD

**Discussion by hannesaddec** [OSGeo/grass#2024] [Q&A] [answered]
Anyone has evaluated GRASS for log4j vulnerability - even its written in C - sometimes it has some knock-on effects ?

**Comment by neteler** [comment]
We are not aware of any connection to the log4j vulnerability.

Do you have any indication?

**Comment by hannesaddec** [comment]
We had a case where in another software a plugin brought in an elastic instance - no further comments on that. Just wondering if anyone has done a full scan.

**Reply by lucadelu** [reply]
In Elastic make sense because it is written in Java, I think GRASS should not be affected.

**Reply by hannesaddec** [reply]
Elastic is one example - it applies for any JAVA based third party module integrated. I don't have the full  overview of all possible extensions/plug-ins which GRASS might have, otherwise I could investigate myself.   
Full scan has been done against source code/compiled version ?

**Comment by wenzeslaus** [comment, marked as answer]
There is no Java or Log4J in the source code:

- https://github.com/OSGeo/grass/search?q=java (9 unrelated results)
- https://github.com/OSGeo/grass/search?q=log4j (no results)

I don't know how to find transitive dependencies for platform-dependent C and C++ libraries, so feel encouraged to contribute in this area.

The code is scanned with GitHub CodeQL which now includes Log4j [CVE-2021-44228](https://github.com/advisories/GHSA-jfh8-c2jp-5v3q) (Remote code injection in Log4j) in question (PRs 7423 and 7354).
